scrapper.py

```diff
@@ -44,20 +82,110 @@ def get_search_urls(self):
+
+        # author
+        table_with_authors = article_bs.find('table', class_='content_table otvet_list')
+        author_list_bs = table_with_authors.find_all('tr')[1]
+        self.article.author = author_list_bs.find('td').text
+
+        # topics
+        meta_info = article_bs.find_all('meta')
+        self.article.topics = meta_info[2]['content']
+
+        # date
+        big_title = article_bs.find('h1')
+        pattern_of_date = r'\d{4}'
+        result = re.findall(pattern_of_date, big_title.text)
+        date_year = int(result[0])
+        self.article.date = datetime.date(date_year, 1, 1)
+
+        # # url
+        # self.article.url = self.article_url
+
+    def _fill_article_with_text(self, article_bs):
+        article_urls_bs = article_bs.find('a', class_='file pdf')
+        pdf = PDFRawFile(article_urls_bs['href'], self.article_id)
+        pdf.download()
+
+        self.article.text = pdf.get_text()
+
+    def parse(self):
+        response = requests.get(url=self.article_url, headers=HEADERS)
+
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path_to_base_path = Path(base_path)
+    if path_to_base_path.exists():
+        shutil.rmtree(base_path)
+    path_to_base_path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        configuration = json.load(file)
+
+    http_pattern = re.compile(HTTP_PATTERN)
+    for url in configuration["seed_urls"]:
+        result = http_pattern.search(url)
+        if not result:
+            raise IncorrectURLError
+
+    seed_urls = configuration["seed_urls"]
+    total_articles_to_find_and_parse = configuration["total_articles_to_find_and_parse"]
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    if not isinstance(total_articles_to_find_and_parse, int):
+        raise IncorrectNumberOfArticlesError
+
+    if total_articles_to_find_and_parse <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if total_articles_to_find_and_parse > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, total_articles_to_find_and_parse
 
 
 if __name__ == '__main__':
-    # YOUR CODE HERE
-    pass
+    print('---Preparing environment---')
+    seed_urls_test, total_articles_test = validate_config(CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+
+    print('---Creating a Crawler---')
+    crawler = Crawler(seed_urls_test, total_articles_test)
+    crawler.find_articles()
+
+    print('---Parsing---')
+    ID_OF_ARTICLE = 0
+    for article_url_test in crawler.urls:
+        ID_OF_ARTICLE += 1
```

Comment:
it is more natural to either increment a counter at the end of a loop which would require changing the starting value

OR finding a more elegant solution of getting the index of current element at each loop iteration - do you remember how?